content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md

```diff
@@ -0,0 +1,312 @@
+  status:
+    conditions:
+    - type: Running
+      status: 'False'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationInProgress
+    - type: Succeeded
+      status: 'True'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationSucceeded
+    resourceVersion: '84'
+  ```
+- [Verify](/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted)
+  the stored secret is now prefixed with `k8s:enc:aescbc:v1:key2`.
+
+## Update the preferred storage schema of the resource while moving from _v1_ to _v2_
+Consider a scenario where a {{< glossary_tooltip term_id="CustomResourceDefinition" text="CustomResourceDefinition" >}}
+(CRD) is created to serve custom resources (CRs) and is set as the preferred storage schema. When it's time
+to introduce v2 of the CRD, it can be added for serving only with a conversion
+webhook. This enables a smoother transition where users can create CRs using
+either the v1 or v2 schema, with the webhook in place to perform the necessary
+schema conversion between them. Before setting v2 as the preferred storage schema
+version, it's important to ensure that all existing CRs stored as v1 are migrated to v2.
+This migration can be achieved through _Storage Version Migration_ to migrate all CRs from v1 to v2.
+
+- Create a manifest for the CRD, named `test-crd.yaml`, as follows:
+  ```yaml
+  apiVersion: apiextensions.k8s.io/v1
+  kind: CustomResourceDefinition
+  metadata:
+    name: selfierequests.stable.example.com
+  spec:
+    group: stable.example.com
+    names:
+      plural: SelfieRequests
+      singular: SelfieRequest
+      kind: SelfieRequest
+      listKind: SelfieRequestList
+    scope: Namespaced
+    versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            hostPort:
+              type: string
+    conversion:
+      strategy: Webhook
+      webhook:
+        clientConfig:
+          url: https://127.0.0.1:9443/crdconvert
+          caBundle: <CABundle info>
+      conversionReviewVersions:
+      - v1
+      - v2
+  ```
+  Create CRD using kubectl
+  ```shell
+  kubectl apply -f test-crd.yaml
+  ```
+- Create a manifest for an example testcrd. Name the manifest `cr1.yaml` and use these contents:
+  ```yaml
+  apiVersion: stable.example.com/v1
+  kind: SelfieRequest
+  metadata:
+    name: cr1
+    namespace: default
+  ```
+  Create CR using kubectl
+  ```shell
+  kubectl apply -f cr1.yaml
+  ```
+- Verify that CR is written and stored as v1 by getting the object from etcd.
+  ```shell
+  ETCDCTL_API=3 etcdctl get /kubernetes.io/stable.example.com/testcrds/default/cr1 [...] | hexdump -C
+  ```
+  where [...] contains the additional arguments for connecting to the etcd server.
+- Update the CRD `test-crd.yaml` to include v2 version for serving and storage
+ and v1 as serving only, as follows:
+  ```yaml
```

Comment:
(nit / OK to skip for alpha)

We can use a `highlight` shortcode to make the changes more obvious; see https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#cleanup-all-secrets-encrypted for an example of what this looks like.